build.go

```diff
@@ -229,6 +229,27 @@ func gitVersion() string {
 	return version
 }
 
+// Constants represents a set constants set in the final binary to the given
+// value.
+type Constants map[string]string
+
+// LDFlags returns the string that can be passed to go build's `-ldflags`.
+func (cs Constants) LDFlags() string {
+	l := make([]string, 0, len(cs))
+
+	if strings.HasPrefix(runtime.Version(), "go1.5") {
+		for k, v := range cs {
+			l = append(l, fmt.Sprintf(`-X "%s=%s"`, k, v))
+		}
+	} else {
+		for k, v := range cs {
+			l = append(l, fmt.Sprintf(`-X %q %q`, k, v))
```

Comment:
Old syntax: `-X "foo" "bar"`, new syntax: `-X "foo=bar"`. In the first case, the quotes are added manually, in the second case `Sprintf()` adds the quotes via %q.